src/common/libflux/message.c

```diff
@@ -1675,7 +1675,25 @@ bool flux_msg_match_route_first (const flux_msg_t *msg1, const flux_msg_t *msg2)
     return true;
 }
 
+bool flux_msg_match_cred (const flux_msg_t *msg1, const flux_msg_t *msg2)
```

Comment:
Would `flux_msg_authrorize()` work here?

Reply:
oh, i had completely missed that you wrote `flux_msg_authorize()`, yeah, that would simplify the function.

Reply:
I didn't know about `flux_msg_authorize()` b/c I saw in `libflux/disconnect.c` that you hand wrote an authorize function.

```
static bool authorize (struct flux_msg_cred cred, const flux_msg_t *msg)
{
    uint32_t userid;

    if ((cred.rolemask & FLUX_ROLE_OWNER))
        return true;
    if ((cred.rolemask & FLUX_ROLE_USER)
        && flux_msg_get_userid (msg, &userid) == 0
        && cred.userid == userid)
        return true;
    return false;
}
```

so perhaps we can clean up some code in `disconnect.c` as well.